File: ShaliniKumari_BanasthaliVidyapith/Arrays/Problem14.cpp
#include <iostream>
#include <vector>
using namespace std;
int findMaxConsecutiveOnes(vector<int>& nums) {
    int ans = 0, c = 0;
    for (int i = 0; i < nums.size(); i++) {
        if (nums[i]) {
            c++;
        } else {
            c = 0;
        }
        ans = max(ans, c);
    }
    return ans;
}
int main() {
    vector<int> nums = {1, 1, 0, 1, 1, 1};
    int maxConsecutiveOnes = solution.findMaxConsecutiveOnes(nums);
    cout << "Max Consecutive Ones: " << maxConsecutiveOnes << endl;
    return 0;
}

File: KMIT/Arrays & Strings assignment SyedShujauddinRafai/Problem14.java
import java.util.*;
public class Problem14 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] nums = new int[n];
        for (int i = 0; i < n; i++) {
            nums[i] = sc.nextInt();
        }
        int maxConsecutiveOnes = findMaxConsecutiveOnes(nums);
        System.out.println(maxConsecutiveOnes);
    }
    public static int findMaxConsecutiveOnes(int[] nums) {
        int maxConsecutiveOnes = 0;
        int n = nums.length;
        int count = 0;
        for (int i = 0; i < n; i++) {
            if (nums[i] == 1) {
                count++;
            }
            else {
                maxConsecutiveOnes = Math.max(maxConsecutiveOnes, count);
                count = 0;
            }
        }
        maxConsecutiveOnes = Math.max(maxConsecutiveOnes, count);
        return maxConsecutiveOnes;
    }
}
